src/client/terminals/codeExecution/helper.ts

```diff
@@ -63,24 +80,69 @@ export class CodeExecutionHelper implements ICodeExecutionHelper {
                     normalizeOutput.resolve(normalized);
                 },
             });
-
+            // If there is no explicit selection, we are exeucting 'line' or 'block'.
+            if (activeEditor?.selection?.isEmpty) {
+                sendTelemetryEvent(EventName.EXECUTION_CODE, undefined, { scope: 'line' });
+            }
             // The normalization script expects a serialized JSON object, with the selection under the "code" key.
             // We're using a JSON object so that we don't have to worry about encoding, or escaping non-ASCII characters.
-            const input = JSON.stringify({ code });
+            const startLineVal = activeEditor?.selection?.start.line ?? 0;
+            const endLineVal = activeEditor?.selection?.end.line ?? 0;
+            const emptyHighlightVal = activeEditor?.selection?.isEmpty ?? true;
+            const smartSendExperimentEnabledVal = pythonSmartSendEnabled(this.serviceContainer)
+                ? pythonSmartSendEnabled(this.serviceContainer)
+                : false;
```

Comment:
```suggestion
            const smartSendExperimentEnabledVal = pythonSmartSendEnabled(this.serviceContainer);
```
Assming `pythonSmartSendEnabled` returns boolean, this ternary check is not needed.

Reply:
@anthonykim1 This is the only thing that I would update.